test/Mono.Linker.Tests.Cases/DataFlow/AnnotatedMembersAccessedViaReflection.cs

```diff
@@ -470,10 +521,26 @@ static void DynamicallyAccessedMembersSuppressedByRUC ()
 			// Duplicated warnings for linker and analyzer see bug https://github.com/dotnet/linker/issues/2462
 			[ExpectedWarning ("IL2111", nameof (AnnotatedProperty.PropertyWithAnnotation) + ".set")]
 			[ExpectedWarning ("IL2111", nameof (AnnotatedProperty.PropertyWithAnnotation) + ".set")]
+			// Linker produces warnings in incorrect places https://github.com/dotnet/linker/issues/2575
+			[ExpectedWarning ("IL2111", nameof (AnnotatedProperty.PropertyWithAnnotation) + ".set", ProducedBy = ProducedBy.Trimmer)]
 			[ExpectedWarning ("IL2111", nameof (AttributeWithPropertyWithAnnotation.PropertyWithAnnotation) + ".set")]
 			[ExpectedWarning ("IL2111", nameof (AttributeWithPropertyWithAnnotation.PropertyWithAnnotation) + ".set")]
 			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotationGetterOnly) + ".get")]
 			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotationGetterOnly) + ".get")]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotation) + ".get")]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotation) + ".get")]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotation) + ".set", ProducedBy = ProducedBy.Analyzer)]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotation) + ".set")]
+			[ExpectedWarning ("IL2111", nameof (PropertyWithAnnotationOnMembers) + ".set")]
+			[ExpectedWarning ("IL2111", nameof (PropertyWithAnnotationOnMembers) + ".set")]
+			[ExpectedWarning ("IL2111", nameof (PropertyWithAnnotationOnMembers) + ".set", ProducedBy = ProducedBy.Trimmer)]
+			[ExpectedWarning ("IL2111", nameof (PropertyWithAnnotationOnMembers) + ".set", ProducedBy = ProducedBy.Trimmer)]
+			[ExpectedWarning ("IL2111", nameof (PropertyWithAnnotationOnMembers) + ".get", ProducedBy = ProducedBy.Trimmer)]
+			[ExpectedWarning ("IL2111", nameof (PropertyWithAnnotationOnMembers) + ".get", ProducedBy = ProducedBy.Trimmer)]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotationOnMembers) + ".get", ProducedBy = ProducedBy.Analyzer)]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotationOnMembers) + ".get", ProducedBy = ProducedBy.Analyzer)]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotationOnMembers) + ".set", ProducedBy = ProducedBy.Analyzer)]
+			[ExpectedWarning ("IL2111", nameof (VirtualPropertyWithAnnotationOnMembers) + ".set", ProducedBy = ProducedBy.Analyzer)]
```

Comment:
I pushed a change which fixes it for the property test - maybe we should do the same for the other tests in this file as well.

The fix is more of a workaround, I simply renamed the properties so that the substrings don't match.
As far as I can tell the correct warnings are generated. And the linker and analyzer behavior matches exactly now (nice).

That said there are still duplicate warnings, but that's a different matter.